### PR TITLE
✨ Bento `amp-adblock-detector`

### LINKED ISSUE #34291: amp-ad: show fallback when ad blocker present
## Show fallback when ad load blocked	

When a user visits a page containing amp-ad elements with an ad blocker enabled, the page should instead show the amp-ad fallback component.

## Additional context

Currently the fallback appears to only work when the ad request fails to retrieve an ad unit for some reason, for exampled no ad is available from the ad manager to fill the request. Instead, the fallback should show when the ad request itself is blocked as is the case when the user is running an ad blocker.

I found some previous work by @sebastianbenz on adblockers in https://github.com/ampproject/amphtml/issues/8311. Reading the amp-ad component documentation, it is unclear whether the current fallback is supposed to work with ad blockers already. In our testing/experience it does not.

### PR DESCRIPTION
**Pull request by AnuragVasanwala**
The goal of the PR is to add Bento AdBlock Detector component for `amp-ad` (for AMP Environment) and other custom advertisement components (for Non-AMP Environment).

This component works as a wrapper component around `amp-ad` or custom advertisement component. When AdBlocker is detected, it will hide advertisement markup and shows fallback.

Fixes #34291

## AMP Environment
- Use `amp-adblock-detector` as a wrapper for `amp-ad` or custom advertisement component
- When AdBlocker is detected, this component hide `amp-ad` (or custom ad component) and shows fallback div (`div[status="blocked"]`)

**Sample (AMP Storybook Environment)**
```html
<amp-adblock-detector layout="fixed" width="120" height="600" {...args}>
  <amp-ad
    width="120"
    height="600"
    type="doubleclick"
    data-slot="/21730346048/test-skyscraper"
  >
    <div fallback>
      <p>Error while loading Ad</p>
    </div>
  </amp-ad>
  <div
    status="blocked"
    style="border: 2px solid red; border-radius: 10px; margin: 5px; padding: 5px;"
  >
    <h2>Ad Blocker Detected</h2>
    <p>Please allow ads to run on this page.</p>
  </div>
</amp-adblock-detector>
```
https://user-images.githubusercontent.com/75766877/146987910-30ebc141-a62d-465a-acd2-2d125760598d.mov

## Preact Environment
- Use `BentoAdblockDetector` as a wrapper
  -  for `amp-ad`, pass it to `ampAd` variable
  - for custom advertisement component, pass them as `children` of `BentoAdblockDetector`
- When AdBlocker is detected, this component hide `ampAd` (or custom ad component inside `children`) and shows `fallbackDiv`

**Sample (Preact Storybook Environment)**
```jsx
<BentoAdblockDetector
  ampAd={(props) => (
    <amp-ad
      width="120"
      height="600"
      type="doubleclick"
      data-slot="/21730346048/test-skyscraper"
      {...props}
    >
      <div fallback>This is AMP-Ad fallback.</div>
    </amp-ad>
  )}
  fallbackDiv={(props) => (
    <div
      style={{
        border: '2px solid red',
        borderRadius: 10,
        margin: 5,
        padding: 5,
      }}
      {...props}
    >
      <h2>Ad Blocker Detected</h2>
      <p>Please allow ads to run on this page.</p>
    </div>
  )}
  style={{width: 120, height: 600}}
  {...args}
></BentoAdblockDetector>
```
https://user-images.githubusercontent.com/75766877/146987997-aa97e275-0a74-4e10-ac89-e98b667ae3bf.mov

### Checklist
- [x] Add `amp-adblock-detector`
- [x] Add types
- [x] Add type annotations
- [ ] Add test case
- [x] Add `npm: true` and `wrapper : "bento"` to `bundles.config.extension.json`
- [x] Update Storybook - AMP and Preact
- [x] latestVersion precedes options
- [ ] Lint, Prettify

### CODE REVIEW

**Comment by AnuragVasanwala** [comment]
⚠️ Need help to write unit-test case to mock AdBlocker behavior by blocking `fetch` request in AMP and Prect unit-test.

File: extensions/amp-adblock-detector/1.0/test/validator-amp-adblock-detector.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
```
Comment (Gregable) [inline extensions/amp-adblock-detector/1.0/test/validator-amp-adblock-detector.html:1]: Please add a copyright statement at the top of this file as well as a description of the test in a comment. See other components for examples.

**Review by Gregable — COMMENTED** [review]
This review is for the technical merits of the validator files (test and protoascii) only. It is not a review of the functionality or purpose of the component itself.

The component likely should also go through a design review as a first step before the pull request stage, if it hasn't already.

File: extensions/amp-adblock-detector/1.0/test/validator-amp-adblock-detector.html (hunk 1 above)
Comment (Gregable) [reply extensions/amp-adblock-detector/1.0/test/validator-amp-adblock-detector.html:1]: Scratch that, this requirement has changed.

**Comment by AnuragVasanwala** [comment]
[✨ Bento amp-adblock-detector #37255](https://github.com/ampproject/amphtml/pull/37255) was presented in design review https://github.com/ampproject/amphtml/issues/37262 and as per the discussion, final conclusion was:
- Find alternative way instead of the component due to [name limitation](https://github.com/ampproject/amphtml/issues/34291#issuecomment-1005972989) of the component.
- It seems quite expensive to maintain component for a generic name because every time we update name, there is a possibility that AdBlock extension may add it to their filter list.

Fortunately, in design review there was a discussion of finding an alternative way of modifying AMP Core to show `amp-ad -> fallback` when script has error or is blocked. It is not exactly as this component but it will serve the purpose.

Closing this PR for now.

**Pull request closed without merging**